### PR TITLE
feat(api): win_viewport also sends line_count

### PR DESCRIPTION
**Pull request by yatli**

### CODE REVIEW

**Comment by bfredl** [comment]
let's only do line count if we extend `win_viewport`, we probably want to start a new event for more buffer info later.

File: src/nvim/api/ui_events.in.h [hunk 1]
```diff
@@ -119,7 +119,8 @@ void msg_set_pos(Integer grid, Integer row, Boolean scrolled, String sep_char)
   FUNC_API_SINCE(6) FUNC_API_BRIDGE_IMPL FUNC_API_COMPOSITOR_IMPL;
 
 void win_viewport(Integer grid, Window win, Integer topline,
-                  Integer botline, Integer curline, Integer curcol)
+                  Integer botline, Integer curline, Integer curcol,
+                  Integer totalline, Integer changedtick)
```
Comment (bfredl) [inline src/nvim/api/ui_events.in.h:123]: `line_count` in analogy with `nvim_buf_get_line_count`

Comment (yatli) [reply src/nvim/api/ui_events.in.h]: done

**Comment by yatli** [comment]
> let's only do line count if we extend win_viewport, we probably want to start a new event for more buffer info later.

okay, updated.

**Comment by bfredl** [comment]
looks good, update `_handle_win_viewport` in `test/functional/ui/screen.lua` to include the new field. Then tests in `multigrid_spec.lua` will need to be updated.

**Comment by yatli** [comment]
@bfredl  ringring 📞

**Comment by clason** [comment]
Isn't that a breaking change to the API? (It broke Neovide, at least ;) -- although I'll have to check whether they relied on undocumented/unpromised behavior.)

**Comment by clason** [comment]
Should also have updated `doc/ui.txt`.

**Comment by bfredl** [comment]
> Isn't that a breaking change to the API?

no, events can append new fields at the end, add new keys to dicts, etc etc, it is all documented in ui.txt. It was merged while I was a sleep so I didn't have the opportunity to review the (non-existant) doc changes :)

**Comment by justinmk** [comment]
> Isn't that a breaking change to the API? (It broke Neovide, at least ;) -- although I'll have to check whether they relied on undocumented/unpromised behavior.)

`:help api-contract`

**Pull request merged**